pkg/controllers/v1alpha1/datatable/datatable_controller.go

```diff
@@ -103,9 +114,63 @@ func (r *DataTableReconciler) reconcileDataTable(ctx reconcileRequestContext) (r
 		return utils.RequeueAfterInterval(10 * time.Second)
 	}
 
+	// 1. create the hive client
+	host := ctx.Datatable.Spec.Url
+	conn, err := hive.CreateHiveClient(host)
+	if err != nil {
+		ctx.Log.Error(err, "Fail to create the hive client")
+		return utils.RequeueIfError(err)
+	}
+	fmt.Println("create the hive client")
+
+	// 2. delete this datatable if it is mark the deletiontimestamp
+	if utils.HasDeletionTimestamp(ctx.Datatable.ObjectMeta) {
+		return r.ReconcileDatatableDeletion(ctx, conn)
+	}
+
+	// 3. add finalizer if it does not have
+	if !utils.ContainsString(ctx.Datatable.ObjectMeta.GetFinalizers(), FinalizerName) {
+		conn.Close()
+		return r.AddFinalizerAndRequeue(ctx)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *DataTableReconciler) ReconcileDatatableDeletion(ctx reconcileRequestContext, conn *gohive.Connection) (ctrl.Result, error) {
+	fmt.Println("delete")
```

Comment:
Better to use r.Log to print debug messages

Reply:
forget to delete this line and i will add the log info